canonicalwebteam/http/heuristics.py

```diff
@@ -0,0 +1,62 @@
+import calendar
+from cachecontrol.heuristics import BaseHeuristic
+from datetime import timedelta, datetime
+from email.utils import formatdate
+
+
+def expire_after(delta, date=None):
+    date = date or datetime.utcnow()
+    return date + delta
+
+
+def datetime_to_header(dt):
+    return formatdate(calendar.timegm(dt.timetuple()))
+
+
+def cache_control_value_present_in_response(headers, type):
+    """
+    checks if a certain CacheControl header is set in the headers
+    """
+    return "cache-control" in headers and type in headers["cache-control"]
+
+
+class ExpiresAfterIfNoCacheControl(BaseHeuristic):
+    """
+    Cache **all** requests for a defined time period.
+    """
+
+    def __init__(self, **kw):
+        self.delta = timedelta(**kw)
+
+    def update_headers(self, response):
+        def cc_value_in_response_headers(type):
```

Comment:
Be careful with naming variable `type` it's a built in python function:

https://stackoverflow.com/questions/10568087/is-it-safe-to-use-the-python-word-type-in-my-code
https://docs.python.org/3/library/functions.html#type